### ISSUE TITLE
Document how to use "Optional Parameters"

### ISSUE THREAD

**Issue by nickrobinson251** [JuliaCloud/AWS.jl#301]
In the docs for the high-level services, e.g. `@service S3`, we list "Required Parameters` and "Optional Parameters"

e.g. here's the docs for `S3.get_object_acl`
https://juliacloud.github.io/AWS.jl/dev/services/s3.html#Main.S3.get_object_acl-Tuple{Any,Any}
![image](https://user-images.githubusercontent.com/13448787/109494368-b02e5e80-7a85-11eb-92af-9620032afecd.png)

**Required Parameters** are passed as postional arguments, e.g. 
```julia
S3.get_bucket_acl(bucket, key)
```
*The docstrings could probably make this clearer too*, but nevertheless, i got this right on my first try, since it's usually for the documented required args to be the positional args

For example, the AWS docstrings closely mirror the common julia style of docstring like
(example taken from Blue Style Guide)
```julia
"""
    mysearch(array::MyArray{T}, val::T; verbose=true) where {T} -> Int

Searches the `array` for the `val`. For some reason we don't want to use Julia's
builtin search :)

# Arguments
- `array::MyArray{T}`: the array to search
- `val::T`: the value to search for

# Keywords
- `verbose::Bool=true`: print out progress details
"""
```

**Optional Parameters** however, it's not clear how to pass at all. Based on the superficial similarity to `Keywords`, you might think Optional Parameters are passed as keywords, like `S3.get_bucket_acl(bucket, key; versionId=version_id)`. (Although unclear what the correct for `x-amz-request-payer` as a keywords would be!) Anyway, this is not the case. 
As far as i understand, "Optional Parameters" are passed as an optional, *positional* argument, as a `Dict{String}` like e.g.
```julia
S3.get_bucket_acl(bucket, key; Dict("versionId" => version_id))
```
Is that right? If so, **i think we should document this.**

---

**How should we document this?**
One option would be to change the docstring signature to be something like
```julia
"""
    get_bucket_acl(required_params...)
    get_bucket_acl(required_params..., optional_params::Dict{String})

...
"""
```

Another would be to be like this but more specific e.g.
```julia
"""
    get_bucket_acl(bucket, key)
    get_bucket_acl(bucket, key, params::Dict{String})

...
"""
```

We may also want to change the headings and text to be like e.g.
```julia
"""
    get_bucket_acl(bucket, key)
    get_bucket_acl(bucket, key, params::Dict{String})

...description...here...

# Arguments
- `bucket`: ...
- `key`: ...

# Optional Parameters
Optional paramters can be passed as a `Dict{String}` with keys:
- `"versionID"`: ...
"""
```

---

**Alternatively, we could provide a more idiomatic Julia interface**, like 
```julia
S3.get_bucket_acl(bucket, key; version_d=1, x_amz_request_payer=2)
```
What do you think?

---
(p.s. the current misleading `GetObjectAcl()` signature will be fixed by #168 / #290)

**Comment by mattBrzezinski** [comment]
Without getting into the weeds again, I think something like the following would be well suited.
One thing to note, some parameters do not have a docstring, they should still be included however.
When AWS decides to add one in, the changes will be picked up.

```julia
"""
    get_bucket_acl(bucket, key)
    get_bucket_acl(bucket, key, optional_params::Dict{String, <:Any})

...description...here...

# Arguments
- `bucket`: ...
- `key`: ...

# Optional Parameters
Optional parameters can be passed as a `Dict{String}` with keys:
- `"versionID"`: ...
"""
```

**Comment by nickrobinson251** [comment]
Cool :) sounds good to me, and i think that should be an easy change to make

**Comment by mattBrzezinski** [comment]
Should just tack into, https://github.com/JuliaCloud/AWS.jl/pull/290

**Comment by nickrobinson251** [comment]
closed by #290
